src/balancerd/src/lib.rs

```diff
@@ -1047,8 +1063,16 @@ impl mz_server_core::Server for HttpsBalancer {
                     .map(|tenant| inner_metrics.tenant_connections(tenant));
 
                 let mut mz_stream = TcpStream::connect(resolved.addr).await?;
+
                 let mut client_counter = CountingConn::new(client_stream);
 
+                // Write the tcp proxy header
+                let addrs = ProxiedAddress::stream(peer_addr, resolved.addr);
+                let header = ProxyHeader::with_address(addrs);
+                let mut buf = [0u8; 1024];
+                let len = header.encode_to_slice_v2(&mut buf)?;
+                mz_stream.write_all(&buf[..len]).await?;
+
```

Comment:
⚠️ I'm pretty sure you need to put this behind a feature flag. balancerd is automatically upgraded when the envd image reference in the stack gets updated. But that happens a few hours before the EU region release and many hours before the US regions release. And so when this change ships, it'll ship to the balancers before envd is capable of handling it.

Reply:
Ok, I think this has been addressed with the new config flag. Tests also pass with the CLI override `--default-config=balancerd_inject_proxy_protocol_header_http=false` 

I'm not sure I like how I've done the cli default config overrides, and may need to revisit before merging.